### PR TITLE
[dynamo] support comparing LHS constant with tensor 

### LINKED ISSUE #108582: [dynamo] scalar <comp op> tensor not supported
Min repro code by @lezcano :

```python
import torch

@torch.compile(fullgraph=True)
def fn(x):
    return 0 <= x

x = torch.randn(512, device="cuda")
fn(x)
```

Here are tracing logs:
```
Step 1: torchdynamo start tracing fn check_compile_le_torch_op.py:4
TRACE starts_line fn check_compile_le_torch_op.py:4
    @torch.compile()
wrap_to_fake L['x'] (512,) [<DimDynamic.STATIC: 2>] [None]
TRACE starts_line fn check_compile_le_torch_op.py:6
        return 0 <= x
TRACE LOAD_CONST 0 []
TRACE LOAD_FAST x [ConstantVariable(int)]
TRACE COMPARE_OP <= [ConstantVariable(int), TensorVariable()]
step triggered compile
Traceback (most recent call last):
  File "/pytorch/torch/_dynamo/symbolic_convert.py", line 691, in step
    getattr(self, inst.opname)(inst)
  File "/pytorch/torch/_dynamo/symbolic_convert.py", line 1107, in COMPARE_OP
    BuiltinVariable(supported_any[op], **options).call_function(
  File "/pytorch/torch/_dynamo/variables/builtin.py", line 618, in call_function
    result = handler(tx, *args, **kwargs)
  File "/pytorch/torch/_dynamo/variables/builtin.py", line 1412, in _comparison
    _unimplemented()
  File "/pytorch/torch/_dynamo/variables/builtin.py", line 1321, in _unimplemented
    unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
  File "/pytorch/torch/_dynamo/exc.py", line 176, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: comparison ConstantVariable(int) <built-in function le> TensorVariable()
restore_graphstate: removed 0 nodes
COMPILING GRAPH due to GraphCompileReason(reason='step_unsupported', user_stack=[<FrameSummary file check_compile_le_torch_op.py, line 6 in fn>], graph_break=True)
GUARDS:
hasattr(L['x'], '_dynamo_dynamic_indices') == False           # _dynamo/variables/builder.py:1252 in wrap_fx_proxy_cls
___is_grad_enabled()                                          # _dynamo/output_graph.py:345 in init_ambient_guards
not ___are_deterministic_algorithms_enabled()                 # _dynamo/output_graph.py:341 in init_ambient_guards
___is_torch_function_enabled()                                # _dynamo/output_graph.py:349 in init_ambient_guards
utils_device.CURRENT_DEVICE == None                           # _dynamo/output_graph.py:347 in init_ambient_guards
check_tensor(L['x'], Tensor, DispatchKeySet(CUDA, BackendSelect, ADInplaceOrView, AutogradCUDA), torch.float32, device=0, requires_grad=False, size=[512], stride=[1])  # _dynamo/variables/builder.py:1252 in wrap_fx_proxy_cls
```

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @ysiraichi

### PR DESCRIPTION
**Pull request by jon-chuang**
Fixes https://github.com/pytorch/pytorch/issues/108582

Depends on https://github.com/pytorch/pytorch/pull/111557 for fixing broken integration tests. (due to this PR unblocking an in-graph set membership)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by jon-chuang** [comment]
opacus_cifar10 is failing. Please investigate.

**Comment by jon-chuang** [comment]
Looks like this approach doesn't handle broadcasting well

```python
import torch

@torch.compile(fullgraph=True)
def fn(x, y):
    return x == y

x = torch.randn(10, device="cuda")
y = torch.randn((10, 512), device="cuda")
print(fn(x, y))
```
torch._dynamo.exc.TorchRuntimeError: Failed running call_function <built-in function eq>(*(FakeTensor(..., device='cuda:0', size=(10,)), FakeTensor(..., device='cuda:0', size=(10, 512))), **{}):
Attempting to broadcast a dimension of length 512 at -1! Mismatching argument at index 1 had torch.Size([10, 512]); but expected shape should be broadcastable to [1, 10]
```

Let's fallback on a less aggressive solution

File: test/test_decomp.py [hunk 1]
```diff
@@ -436,6 +436,7 @@ def test_unsupported(t):
     skip('_softmax_backward_data'),  # slow: fails with --timeout=360 secs
     xfail('addcdiv'),
     skip('addcmul'),  # slow: fails with --timeout=360 secs
+    skip('baddbmm'),  # slow: takes 800+ sec on A100
```
Comment (jon-chuang) [inline test/test_decomp.py:439]: remove stray

File: torch/_dynamo/variables/lists.py [hunk 2]
```diff
@@ -34,18 +34,38 @@ def _listlike_contains_helper(items, search, tx, options):
 
     from .builtin import BuiltinVariable
 
-    result = None
+    must_check_hash = False
```
Comment (jon-chuang) [inline torch/_dynamo/variables/lists.py:37]: change to `check_object_match`

File: test/dynamo/test_misc.py [hunk 3]
```diff
@@ -1466,6 +1466,9 @@ def fn(op, t1, t2):
         for op, t1_np, t2_np in itertools.product(
             operators, (True, False), (True, False)
         ):
+            if op in [operator.ne, operator.eq]:
+                # This always returns single boolean which says they are not equal
+                continue
```
Comment (jon-chuang) [inline test/dynamo/test_misc.py:1471]: remove stray

File: test/dynamo/test_functions.py [hunk 4]
```diff
@@ -1835,6 +1835,42 @@ def fn(x):
         ref = opt_fn(x)
         self.assertEqual(ref, res)
 
+    def test_broadcast_in_set(self):
```
Comment (jon-chuang) [inline test/dynamo/test_functions.py:1838]: + test_broadcast_error_in_set

**Pull request closed without merging**